controllers/dynamic_resource_request_controller.go

```diff
@@ -103,30 +98,36 @@ func (r *dynamicResourceRequestController) Reconcile(ctx context.Context, req ct
 		return addFinalizers(ctx, r.Client, rr, []string{workFinalizer, workflowsFinalizer, deleteWorkflowsFinalizer}, logger)
 	}
 
-	if !r.hasCondition(PipelineCompletedCondition, rr) {
-		r.setStatus(rr, logger, "message", "Pending")
-		r.setCondition(clusterv1.Condition{
-			Type:               PipelineCompletedCondition,
-			Status:             v1.ConditionFalse,
-			Message:            "Pipeline has not completed",
-			Reason:             "PipelineNotCompleted",
-			LastTransitionTime: metav1.NewTime(time.Now()),
-		}, rr, logger)
-		return ctrl.Result{}, r.Client.Status().Update(ctx, rr)
+	pipelineJobs, err := r.getPipelineJobs(resourceRequestIdentifier, rr.GetNamespace(), logger)
+	if err != nil {
+		logger.Info("Failed getting pipeline jobs", "error", err)
+		return slowRequeue, nil
 	}
 
-	//check if the pipeline has already been created. If it has exit out. All the lines
-	//below this call are for the one-time creation of the pipeline.
-	created, err := r.configurePipelineHasBeenCreated(resourceRequestIdentifier, rr.GetNamespace(), logger)
-	if err != nil {
-		return ctrl.Result{}, err
+	if len(pipelineJobs) != 0 {
+		if resourceutil.IsThereAPipelineRunning(logger, pipelineJobs) {
+			return slowRequeue, nil
+		}
+
+		if !resourceutil.IsAnUpdate(logger, rr, pipelineJobs) {
+			return ctrl.Result{}, nil
+		}
 	}
 
-	if created {
-		logger.Info("Cannot execute update on pre-existing pipeline for Promise resource request " + resourceRequestIdentifier)
-		return ctrl.Result{}, nil
+	if resourceutil.LastPipelineCompleted(rr) {
+		r.setStatus(rr, logger, "message", "Update Pending")
+		resourceutil.MarkPipelineAsRunning(logger, rr)
+		return ctrl.Result{}, r.Client.Status().Update(ctx, rr)
 	}
 
+	if !resourceutil.HasPipelineCompletedCondition(rr) {
+		r.setStatus(rr, logger, "message", "Pending")
+		resourceutil.MarkPipelineAsRunning(logger, rr)
+		return ctrl.Result{}, r.Client.Status().Update(ctx, rr)
+	}
+
+	logger.Info("Triggering pipeline", "resourceRequest", resourceRequestIdentifier)
```

Comment:
these two checks are making my head hurt 😅 

is the 1st more `if PipelineIsNotMarkedasUpdatePending; then; markAsUpdatePending; done`

unsure what the 2nd is, particularly in the context of the first 🤔

Reply:
see above